### PR TITLE
Refactor recommended sites redux store to TypeScript

### PR DESCRIPTION
**Pull request by mehmoodak**
Related to https://github.com/Automattic/loop/issues/233

## Proposed Changes

* Refactors recommended sites store to TypeScript.
* Refactors reader `FollowingStream` and `ReaderPopularSitesSidebar` components to TypeScript.

## Why are these changes being made?

* To improve typing in codebase.

## Testing Instructions

* No change in functionality. Make sure reader works as expected.

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/reader/stream/reader-popular-sites-sidebar/index.tsx [hunk 1]
```diff
@@ -1,28 +1,58 @@
...
+
 	return {
 		feed_ID: item.feed_ID,
 		blog_ID: item.blogId,
 		URL: item.feed_URL ?? item.url,
 		name: unescape( item?.site_name ),
 		site_icon: item.site_icon ?? null,
 		description: unescape( item?.site_description ),
-		last_updated: 0,
-		unseen_count: 0,
```
Comment (mehmoodak) [inline client/reader/stream/reader-popular-sites-sidebar/index.tsx:21]: Not used anywhere.

File: client/state/data-layer/wpcom/read/recommendations/sites/index.ts
Comment (mehmoodak) [inline client/state/data-layer/wpcom/read/recommendations/sites/index.ts:1]: Renamed an existing file to `*.ts` and added interfaces.

File: client/state/data-layer/wpcom/read/recommendations/sites/test/index.ts [hunk 2]
```diff
@@ -3,35 +3,74 @@ import {
...
+			description: 'Description 1',
+			feed_id: 185124,
+			feed_url: 'http://bentehaarstad.wordpress.com/feed/',
+			icon: {
+				ico: 'http://bentehaarstad.wordpress.com/favicon.ico',
+				img: 'http://bentehaarstad.wordpress.com/favicon.ico',
+				media_id: '12345',
+			},
+			ID: 12345,
+			name: 'Bente Haarstad Photography&amp;',
```
Comment (mehmoodak) [inline client/state/data-layer/wpcom/read/recommendations/sites/test/index.ts:32]: Adding missing response fields to ensure that our implementation matches with the actual API.

File: client/state/data-layer/wpcom/read/recommendations/sites/test/index.ts [hunk 3]
```diff
@@ -55,20 +94,21 @@ describe( 'recommended sites', () => {
...
+					sites: recommendedSites,
 					seed,
 					offset: 0,
 				} )
 			);
 		} );
 	} );
 
-	describe( '#fromApi', () => {
+	describe( '#mapResponseToRecommendedSites', () => {
```
Comment (mehmoodak) [inline client/state/data-layer/wpcom/read/recommendations/sites/test/index.ts:109]: Just improved name of the function.

File: client/state/reader/recommended-sites/types.ts
Comment (mehmoodak) [inline client/state/reader/recommended-sites/types.ts:1]: Moved the only interface in this file to `client/state/data-layer/wpcom/read/recommendations/sites/index.ts`

File: client/state/reader/recommended-sites/selectors/get-reader-recommended-sites-paging-offset.ts
Comment (mehmoodak) [inline client/state/reader/recommended-sites/selectors/get-reader-recommended-sites-paging-offset.ts:1]: Rename the file from `.js` to `.ts`.

File: client/state/reader/recommended-sites/selectors/get-reader-recommended-sites.ts
Comment (mehmoodak) [inline client/state/reader/recommended-sites/selectors/get-reader-recommended-sites.ts:1]: Rename the file from `.js` to `.ts`.

File: client/reader/following/main.tsx [hunk 4]
```diff
@@ -7,26 +7,24 @@ import NavigationHeader from 'calypso/components/navigation-header';
...
 import { useFollowingView } from './view-preference';
 import ViewToggle from './view-toggle';
 import './style.scss';
 
 function FollowingStream( { ...props } ) {
-	const { currentView, setView } = useFollowingView();
+	const { currentView } = useFollowingView();
 
-	const viewToggle = config.isEnabled( 'reader/recent-feed-overhaul' ) ? (
-		<ViewToggle currentView={ currentView } onChange={ setView } />
```
Comment (mehmoodak) [inline client/reader/following/main.tsx:21]: The `currentView` and `onChange` props are not needed.

**Review by mehmoodak — COMMENTED** [review]
Added some comments for easier review.

**Review by holdercp — APPROVED** [review]
Thanks for this refactor! Types look good and I didn't find any regressions.

**Pull request merged**